### ISSUE TITLE
iOS App Crashes when using Chimera Jailbreak

### ISSUE THREAD

**Issue by TwoTenPvP** [bitwarden/mobile#493]
Running the Bitwarden app on a Rooted/Jailbroken iOS device causes it to crash. This cannot be reproduced with the [unc0ver](https://github.com/pwn20wndstuff/Undecimus) jailbreak, but it does happen consistently with the [Chimera](https://chimera.sh/) jailbreak version 1.0.2 (latest at the time of creating)

**Comment by kspearrin** [comment]
Unfortunately, I do not have any experience with jailbreaking iOS devices so I am not sure how I could debug this issue.

**Comment by TwoTenPvP** [comment]
> Unfortunately, I do not have any experience with jailbreaking iOS devices so I am not sure how I could debug this issue.

I completely understand.. It's a niche user market but one that I am a part of. I truley love Bitwarden and would not want to switch from it. If there is anything I can do to assist debugging the problem let me know.

(I am an experienced C# developer and experienced android developer, but not iOS developer)

**Comment by kspearrin** [comment]
This is a Xamarin mobile app, written in C#, so you could try debugging a dev build locally in visual studio with your device.

**Comment by TwoTenPvP** [comment]
> This is a Xamarin mobile app, written in C#, so you could try debugging a dev build locally in visual studio with your device.

Right, the only problem is that I don't own a macintosh. I'll throw a KVM and see where it takes me, hopefully I can throw some stacktraces your way or even PR it..

**Comment by kspearrin** [comment]
I see. Unfortunately, I don't think that it will work if you don't have a MacBook. I have never tried it.

**Comment by TwoTenPvP** [comment]
> I see. Unfortunately, I don't think that it will work if you don't have a MacBook. I have never tried it.

I've had great success with MacOs kvm's for CI purposes previously with Xamarin, I will try to get back to you. Thanks for this beautiful password manager.

**Comment by Nepeta** [comment]
@TwoTenPvP @kspearrin 

Hey, I'm trying to debug the issue and I've also contacted the Electra Team, providing them with crash logs.

This issue only should happen on A12(X) (arm64e) devices running the Chimera jailbreak, and it's related to newly introduced PAC (on arm64e).

I haven't checked yet if the app itself is compiled with arm64e slices in the binaries. I'm trying to compile it with just arm64 but can't get the build to work on my High Sierra machine. I'm using Visual Studio for Mac. If someone could me provide with a build of the app (in a form of an .ipa file) that only contains arm64 binaries (not armv7, and certainly not arm64e) I could give it a try and report back.

Here are the relevant parts of the crash log: (note: stack traces are entirely useless here, so I didn't bother pasting them)

```
Exception Type:  EXC_BAD_ACCESS (SIGKILL)
Exception Subtype: EXC_ARM_DA_ALIGN at 0x0040002f7261762f -> 0x0000002f7261762f (possible pointer authentication failure)
VM Region Info: 0x2f7261762f is not in any region.  Bytes after previous region: 192508163632  
      REGION TYPE                      START - END             [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      MALLOC_NANO            0000000280000000-00000002a0000000 [512.0M] rw-/rwx SM=PRV  
--->  
      UNUSED SPACE AT END

Termination Reason: Namespace SPRINGBOARD, Code 0x8badf00d
Termination Description: SPRINGBOARD, scene-create watchdog transgression: com.8bit.bitwarden exhausted real (wall clock) time allowance of 19.92 seconds | ProcessVisibility: Foreground | ProcessState: Running | WatchdogEvent: scene-create | WatchdogVisibility: Foreground | WatchdogCPUStatistics: ( | "Elapsed total CPU time (seconds): 29.400 (user 29.400, system 0.000), 24% CPU", | "Elapsed application CPU time (seconds): 19.984, 17% CPU" | )
Triggered by Thread:  0
```

**Comment by Nepeta** [comment]
After some testing I've confirmed this issue is limited only to arm64e devices.

iPhone 7+ (11.3.1; latest Electra [which is similar in architecture to Chimera]) - app loads without issues
iPhone SE (12.1.2; latest Chimera) - app loads without issues

The issue may be with the pspawn payload used by Chimera for arm64e.

**Comment by kspearrin** [comment]
We compile the app only for ARM64 in release mode.

**Comment by Nepeta** [comment]
@kspearrin I've just noticed taking the look on the binaries loaded in the crash log. The error message is probably irrelevant though. I'll look for the real issue.

**Comment by Flipp3rrr** [comment]
@Nepeta An error message should be useful, maybe someone knows what the message means. But this issue is only on an A12 chipset not on the iPhone 6's A8?

**Comment by Flipp3rrr** [comment]
If needed I can use my MacBook and iPhone 6 for debugging too.

**Comment by Nepeta** [comment]
The issue should be resolved in an upcoming version of Chimera.

**Comment by Flipp3rrr** [comment]
Ok. Thanks.

**Comment by TwoTenPvP** [comment]
This issue is resolved in Chimera 1.0.6
